File: common/src/main/java/net/impleri/playerskills/integration/kubejs/skills/TieredSkillJS.java
package net.impleri.playerskills.integration.kubejs.skills;

import net.impleri.playerskills.api.TeamMode;
import net.impleri.playerskills.variant.tiered.TieredSkill;
import net.minecraft.resources.ResourceLocation;

public class TieredSkillJS extends TieredSkill {
    public TieredSkillJS(Builder builder) {
        super(builder.id, builder.options, builder.initialValue, builder.description, builder.changesAllowed, builder.teamMode, builder.notify, builder.notifyKey);
    }

    public static class Builder extends GenericSkillBuilderJS<String> {

        public Builder(ResourceLocation name) {
            super(name);
        }

        @Override
        public TieredSkillJS createObject() {
            return new TieredSkillJS(this);
        }

        public Builder pyramid() {
            teamMode = TeamMode.pyramid();

            return this;
        }
    }
}

File: common/src/main/kotlin/net/impleri/playerskills/integrations/kubejs/skills/TieredSkillJS.kt
package net.impleri.playerskills.integrations.kubejs.skills

import net.impleri.playerskills.api.Skill
import net.impleri.playerskills.api.TeamMode
import net.impleri.playerskills.skills.tiered.TieredSkill
import net.minecraft.resources.ResourceLocation

class TieredSkillJS(builder: Builder) : TieredSkill(
  builder.id,
  builder.options,
  builder.initialValue,
  builder.description,
  builder.changesAllowed,
  builder.teamMode,
  builder.notify,
  builder.notifyKey,
) {
  class Builder(name: ResourceLocation) : GenericSkillBuilderJS<String>(name) {
    override fun createObject(): Skill<String> {
      return TieredSkillJS(this)
    }

    fun pyramid(): Builder {
      teamMode = TeamMode.pyramid()
      return this
    }
  }
}
